Mobile: Add accessibility label to Block List Footer

shiki opened the pull request:
I worked with @jd-alexander on this during Gutenberg Day. 🙂 

## Description

Please see the corresponding Mobile PR in https://github.com/wordpress-mobile/gutenberg-mobile/pull/1981 for the details.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

shiki committed:
Make Block List Footer inaccessible

shiki committed:
Add a11y label to Block List Footer

shiki committed:
Fix lint issue

etoledom reviewed (commented):
Thank you @shiki and @jd-alexander for this PR!

This is working perfectly as described, but there's a small detail:

From https://github.com/wordpress-mobile/gutenberg-mobile/pull/1981#issue-383903658

> it looks like that element is used for adding a block at the bottom while the Add Block button on the toolbar is used for adding a block next to the currently selected block.

This is only partially true: The `add block` button will also append a block at the end if no block is selected, but more importantly, tapping in that area will only append a `Paragraph Block`. I believe it is important to signal this to the user. (i.e: "Add paragraph block" or "Add new paragraph block here")

shiki committed:
Change Block List Footer accessibility label

Adding “paragraph” would make it clearer and disambiguates it from the “Add block” button which behaves differently (shows a popup).

shiki committed:
Merge branch master into issue/make-add-block-inaccessible

Tug commented on packages/block-editor/src/components/block-list/index.native.js:
```diff
@@ -23,6 +23,7 @@ import styles from './style.scss';
 import BlockListBlock from './block';
 import BlockListAppender from '../block-list-appender';
 import BlockInsertionPoint from './insertion-point';
+import { __ } from '@wordpress/i18n';
```
This should go into the `WordPress dependencies` section.

shiki replied:
Thanks! Didn't notice that. Fixed in f0f6ce3.

shiki committed:
block-list: Move import to correct section

shiki commented:
Thank you, @etoledom and @Tug. I have applied the requested changes. I changed the accessibility label to "Add paragraph block" and reorganized the import. 

Ready for another look! 

---

I'm not sure what to do with https://github.com/wordpress-mobile/gutenberg-mobile/pull/1981. Should I update that only when this is merged?

etoledom commented:
> I'm not sure what to do with wordpress-mobile/gutenberg-mobile#1981. Should I update that only when this is merged?

@shiki - We usually keep updating the `gutenberg` reference with the latest commit from this PR for testing purposes. When this PR gets merged, we update the`gutenberg-mobile` PR with the merge commit from `gutenberg:master` and merge.

Another tip to facilitate review and testing, we can temporarily edit `.gitmodules` to point to your fork, so a simple `git submodule update` will work for us. Then we reverse that change before merging.

shiki commented:
Looks like there are legitimate test failures in wordpress-mobile/gutenberg-mobile#1981. I'm gonna take a look at that.

etoledom commented:
> Looks like there are legitimate test failures in [wordpress-mobile/gutenberg-mobile#1981](https://github.com/wordpress-mobile/gutenberg-mobile/pull/1981). I'm gonna take a look at that.

I had a similar problem, and got solved by revising the `gutenberg` PR with latests master and merging `gutenberg-mobile` from latest develop. Not sure if it's related or if it was just CI momentarily misbehaving.

etoledom reviewed (approved):
Thanks for the updates, looks great! 🎉 

I re-triggered CI on https://github.com/wordpress-mobile/gutenberg-mobile/pull/1981 and now they are ✅
